source/material_model/steinberger.cc

```diff
@@ -375,24 +406,39 @@ namespace aspect
           equation_of_state.initialize_simulator (this->get_simulator());
           equation_of_state.parse_parameters(prm);
 
+          // Check if compositional fields represent a composition
+          std::vector<typename Parameters<dim>::CompositionalFieldDescription> composition_descriptions = this->introspection().get_composition_descriptions();
+          unsigned int n_chemical_fields = 0;
+
+          // All chemical compositional fields are assumed to represent mass fractions.
+          // If the field type is unspecified (has not been set in the input file),
+          // we have to assume it also represents a chemical composition for reasons of
+          // backwards compatibility.
+          composition_mask = std::make_unique<ComponentMask> (this->n_compositional_fields(), false);
+          for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+            if (composition_descriptions[c].type == Parameters<dim>::CompositionalFieldDescription::chemical_composition
+                || composition_descriptions[c].type == Parameters<dim>::CompositionalFieldDescription::unspecified)
+              {
+                composition_mask->set(c, true);
+                n_chemical_fields++;
```

Comment:
```suggestion
                ++n_chemical_fields;
```
but I think you can use ``n_chemical_fields = composition_mask->n_selected_components();`` at the end.

Reply:
Thanks! I was wondering how to do that.